File: functions/libsodium/variants/c/0_b64_byte_to_char_1.c
#include <stdint.h>

static int b64_byte_to_char(unsigned int x)
{
    return x < 26 ? x + 'A' : (x < 52 ? x - 26 + 'a' : (x < 62 ? x - 52 + '0' : (x == 62 ? '+' : '/')));
}


int main(){
    b64_byte_to_char(0);
}

File: functions/libsodium/variants/go/0_b64_byte_to_char_1.go
package main

func b64ByteToChar(x uint32) int32 {
    switch {
    case x < 26:
        return int32(x + 'A')
    case x < 52:
        return int32(x - 26 + 'a')
    case x < 62:
        return int32(x - 52 + '0')
    case x == 62:
        return '+'
    default:
        return '/'
    }
}


func main(){
    b64ByteToChar(0)
}
